native/core/src/execution/shuffle/shuffle_writer.rs

```diff
@@ -314,65 +356,45 @@ impl ShuffleRepartitioner {
         codec: CompressionCodec,
         enable_fast_encoding: bool,
     ) -> Result<Self> {
-        let mut hashes_buf = Vec::with_capacity(batch_size);
-        let mut partition_ids = Vec::with_capacity(batch_size);
-
-        // Safety: `hashes_buf` will be filled with valid values before being used.
-        // `partition_ids` will be filled with valid values before being used.
-        unsafe {
-            hashes_buf.set_len(batch_size);
-            partition_ids.set_len(batch_size);
-        }
+        let num_output_partitions = partitioning.partition_count();
```

Comment:
Is `assert_ne(num_output_partitions, 1, "Use SinglePartitionShufflePartitioner for 1 output partition.")` a valid assertion to add?

Reply:
Added the assertion.